core/include/tee/uuid.h

```diff
@@ -23,4 +27,16 @@ void tee_uuid_to_octets(uint8_t *dst, const TEE_UUID *src);
  */
 void tee_uuid_from_octets(TEE_UUID *dst, const uint8_t *src);
 
+/**
+ * tee_uuid_from_string() - de-serialize a string like
+ * 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' into a TEE_UUID
+ * @dst: pointer TEE_UUID
+ * @str: pointer to a string
+ * @len: length of the string
+ *
+ * Returns TEE_SUCCESS, when successfully converting a string or an error code,
+ * if it has a wrong format or NULL.
+ */
+TEE_Result tee_uuid_from_string(TEE_UUID *dst, const char *str, int len);
```

Comment:
wow, I will try to use the  `tee_uuid_from_str()` from `libutee` :)
In this case I will drop own implementation.

Reply:
As I said even better would be to not use character strings at all and use a `TEE_UUID` instead, unless I am missing something?

Reply:
TAs work with plugins by its names and that's why the TAs pass the string with name to `TEE_InvokeTACommand()`, like [here](https://github.com/linaro-swg/optee_examples/pull/79/commits/66cacca60e8aa5360679e4a39485ac28fbe95997#diff-2bcac3f86cda44f4c03fdc5aeace96f093f4ad9f9f59b87f9ebe5d7a045ff13aR97).  This string is passed to the plugin-pta. 
We can use `tee_uuid_from_str()`  inside the PTA before calling `tee_invoke_plugin_rpc()`.
If we will not use character strings at all in optee, we will need convert the strings in TA, when we want to call plugins.
I don't mind, but seems it looks not convinient for TA developes. What do you think?

Reply:
You could do:
```C
#define SYSLOG_PLUGIN_UUID { 0x96bcf744, 0x4f72, 0x4866, \
                         { 0xbf, 0x1d, 0x86, 0x34, 0xfd, 0x9c, 0x65, 0xe5 } }
```
...instead of:
```C
#define SYSLOG_PLUGIN_NAME "96bcf744-4f72-4866-bf1d-8634fd9c65e5"
```
That's how TAs open other TAs or PTAs for example. The only place you may need to convert from binary UUIDs to a string is in tee-supplicant when looking for a plugin (and even in this case it may not be necessary, I mean assuming tee-supplicant enumerates and loads all the plugins on startup, it would convert from plugin name to UUID instead, or find the UUID in the plugin).

Reply:
OK, got it, thanks!